drafts/proof-of-spacetime.md

```diff
@@ -1,24 +1,141 @@
 ## Proof-of-Spacetime
-   
-This document descibes 
- 
+
+This document descibes
+
+- Base-PoSt
 - VDF-PoSt: a Proof-of-Spacetime using VDFs
 - An Extension to PoSt to support multiple sectors
-- An Extension to PoSt to support challenges taken from a Random Beacon 
+- An Extension to PoSt to support challenges taken from a Random Beacon
 
 ## Syntax
 
 - **PoSt Epoch**: The total time passing between Online PoReps in the PoSt Computation (in VDF, this interval is the time it takes to run a VDF and an Online PoRep prove step). We define the number of epochs as `POST_EPOCHS_COUNT`
-- **PoSt Period**: The total time it takes to run a single PoSt. If a PoSt is repeated multiple times, we define the number of periods as `POST_PERIODS_COUNT`. This can be reasoned as: `PoSt Epoch * POST_EPOCHS_COUNT`. We assume that the best  Post Period time is `MIN_POST_PERIOD_TIME` 
+- **PoSt Period**: The total time it takes to run a single PoSt. If a PoSt is repeated multiple times, we define the number of periods as `POST_PERIODS_COUNT`. This can be reasoned as: `PoSt Epoch * POST_EPOCHS_COUNT`. We assume that the best  Post Period time is `MIN_POST_PERIOD_TIME`
 - **Total proving time**: The time it takes to run a PoSt. Note that a PoSt could be a composition of multiple PoSt. This can be reasoned as: `PoSt Period * POST_PERIODS_COUNT` TODO Check with papers' syntax
 
+
+## Base-PoSt
+
+- **Setup Parameters**
+  - Same as Public parameters.
+- **Public Parameters**
+  - `post_periods_count`: uint
+  - `challenge_count`: number of challenges to be asked at each iteration
+  - `sector_size`: size of the sealed sector in bytes
+  - `sectors_count`: number of sectors over which the proof is performed
```

Comment:
Here is a practical question: what is the concrete plan for allowing miners to flexibly prove over many sectors without submitting multiple posts to the chain or running two posts at once?

Is it simply to generate a (`pk`, `vk`) pair for every circuit with sector count {1, 2, 3, 4, 5, ... Max Sectors per miner}, and then have provers and verifiers use the correct key during proof generation and verification?  

Since we're probably talking about gigabytes per key what is the feasible limit for distinct sectors count circuits?

Reply:
I'm now remember something about being able to pass in subtrees of zeros as private inputs along with a sector's count to a single circuit.  Does that sound right?

Reply:
@ZenGround0 

> what is the concrete plan for allowing miners to flexibly prove over many sectors without submitting multiple posts to the chain or running two posts at once?

@nicola had the idea (and I [have put up a PR to match](https://github.com/filecoin-project/specs/pull/348)) to cap the number of sectors which a single miner actor can commit to the network to 16384 (`POST_SECTORS_COUNT`, or 1PiB/64GiB sectors). This number forms a part of the PoSt's public inputs, and specifies the number of sectors for which a single PoSt will be generated. If a single miner operator wishes to commit more than 16384 sectors to the network, they'll have to provision more than one storage miner actor (and generate multiple, _independent_ PoSts in parallel).

Note that the aforementioned PR has not yet been ratified into the spec.

Reply:
as @laser said there will be a maximum of sectors that can be submitted per post (and which will be the limit for a single storage miner actor). Details on how this will work exactly will be added into this doc

Reply:
there is no maximum in the latest construction, all sectors can be proving in a single PoSt